python/pyspark/sql/streaming/state_api_client.py

```diff
@@ -0,0 +1,172 @@
+            schema = cast(StructType, _parse_datatype_string(schema))
+        print(f"updating value state: {state_name}")
+        byteStr = value.encode('utf-8')
+        update_call = stateMessage.Update(stateName=state_name, schema=schema.json(), value=byteStr)
+        value_state_call = stateMessage.ValueStateCall(update=update_call)
+        state_variable_request = stateMessage.StateVariableRequest(valueStateCall=value_state_call)
+        message = stateMessage.StateRequest(stateVariableRequest=state_variable_request)
+
+        self._send_proto_message(message)
+        status = read_int(self.sockfile)
+        print(f"valueStateUpdate status= {status}")
+
+    def value_state_clear(self, state_name: str) -> None:
+        print(f"clearing value state: {state_name}")
+        clear_call = stateMessage.Clear(stateName=state_name)
+        value_state_call = stateMessage.ValueStateCall(clear=clear_call)
+        state_variable_request = stateMessage.StateVariableRequest(valueStateCall=value_state_call)
+        message = stateMessage.StateRequest(stateVariableRequest=state_variable_request)
+
+        self._send_proto_message(message)
+        status = read_int(self.sockfile)
+        print(f"valueStateClear status= {status}")
+
+    def set_implicit_key(self, key: str) -> None:
+        print(f"setting implicit key: {key}")
+        set_implicit_key = stateMessage.SetImplicitKey(key=key)
+        request = stateMessage.ImplicitGroupingKeyRequest(setImplicitKey=set_implicit_key)
+        message = stateMessage.StateRequest(implicitGroupingKeyRequest=request)
+
+        self._send_proto_message(message)
+        status = read_int(self.sockfile)
+        print(f"setImplicitKey status= {status}")
+
+    def remove_implicit_key(self) -> None:
+        print(f"removing implicit key")
+        remove_implicit_key = stateMessage.RemoveImplicitKey()
+        request = stateMessage.ImplicitGroupingKeyRequest(removeImplicitKey=remove_implicit_key)
+        message = stateMessage.StateRequest(implicitGroupingKeyRequest=request)
+
+        self._send_proto_message(message)
+        status = read_int(self.sockfile)
+        print(f"removeImplicitKey status= {status}")
+
+    def _get_proto_state(self,
+                         state: StatefulProcessorHandleState) -> stateMessage.HandleState.ValueType:
+        if (state == StatefulProcessorHandleState.CREATED):
+            return stateMessage.CREATED
+        elif (state == StatefulProcessorHandleState.INITIALIZED):
+            return stateMessage.INITIALIZED
+        elif (state == StatefulProcessorHandleState.DATA_PROCESSED):
+            return stateMessage.DATA_PROCESSED
+        else:
+            return stateMessage.CLOSED
+
+    def _send_proto_message(self, message: stateMessage.StateRequest) -> None:
+        serialized_msg = message.SerializeToString()
+        print(f"sending message -- len = {len(serialized_msg)} {str(serialized_msg)}")
+        write_int(0, self.sockfile)
```

Comment:
Why do we need to send a zero int explicitly first?

Reply:
could we add a comment here if we intend to keep this ?